### PR TITLE
cursor: allow acting as a transducer

### PR DESCRIPTION
**Pull request by pyr**

### CODE REVIEW

**Review by arnaudgeiser — APPROVED** [review]
Nice, thank you!

File: src/exoscale/vinyl/cursor.clj [hunk 1]
```diff
@@ -32,18 +35,24 @@
...
                      (and (not (reduced? new-acc)) next?))))))))
        (.getExecutor cursor))
-      (fn/make-fun (fn [_] (unreduced @acc))))))
+      (fn/make-fun (fn [_]
+                     (unreduced
+                      (cond-> @acc
+                        (some? xform)
+                        reducer)))))))
+  ([cursor f init cont-fn]
+   (apply-reduce cursor nil f init cont-fn))
```
Comment (mping-exo) [inline src/exoscale/vinyl/cursor.clj]: Shouldn't `apply-reduce` be renamed to `apply-transduce`, even if sometimes its just reducing?

File: src/exoscale/vinyl/store.clj [hunk 2]
```diff
@@ -458,10 +458,10 @@
    ::max-delay           2
    ::initial-delay       2})
 
-(defn continuation-traversing-reducer
-  "A reducer over large ranges.
+(defn continuation-traversing-transducer
```
Comment (mping-exo) [inline src/exoscale/vinyl/store.clj]: Nitpick: `continuation-traversing-transducer` does an actual transduce, `long-range-reducer` does an actual reduction, Shouldn't they be called `continuation-traversing-transduce` and `long-range-reduce`, similar to `core/transduce` and `core/reduce`?

**Comment by mping-exo** [comment]
Really cool stuff, just added some minor comments

File: src/exoscale/vinyl/store.clj (hunk 2 above)
Comment (pyr) [reply src/exoscale/vinyl/store.clj]: @arnaudgeiser would you be ok with the name change?

**Review by mping-exo — APPROVED** [review]

**Pull request merged**